tests/app/main/views/test_suppliers.py

```diff
@@ -368,6 +368,181 @@ def test_should_show_separate_tables_for_frameworks_if_supplier_has_service_on_f
+
+    @pytest.mark.parametrize('service_status', ['enabled', 'disabled', 'a_new_status'])
+    @mock.patch('app.main.views.suppliers.data_api_client')
+    def test_400_if_supplier_has_no_published_service_on_framework(self, data_api_client, service_status):
+        framework = 'g-cloud-8'
+        service = self.load_example_listing('services_response')['services'][0]
+        service["status"] = service_status
+
+        data_api_client.get_supplier.return_value = self.load_example_listing('supplier_response')
+        data_api_client.find_services.return_value = {'services': [service]}
+
+        response = self.client.get('/admin/suppliers/1000/services?remove={}'.format(framework))
+
+        assert response.status_code == 400
+
+    @mock.patch('app.main.views.suppliers.data_api_client')
+    def test_200_if_supplier_has_published_service_on_framework(self, data_api_client):
+        framework = 'g-cloud-8'
+
+        data_api_client.get_supplier.return_value = self.load_example_listing('supplier_response')
+        data_api_client.find_services.return_value = self.load_example_listing('services_response')
+        data_api_client.find_frameworks.return_value = {
+            'frameworks': [self.load_example_listing("framework_response")['frameworks']]
+        }
+
+        response = self.client.get('/admin/suppliers/1000/services?remove={}'.format(framework))
+
+        assert response.status_code == 200
+
+    @mock.patch('app.main.views.suppliers.data_api_client')
+    def test_are_you_sure_banner_if_supplier_has_published_service_on_framework(self, data_api_client):
+        framework = 'g-cloud-8'
+
+        data_api_client.get_supplier.return_value = self.load_example_listing('supplier_response')
+        data_api_client.find_services.return_value = self.load_example_listing('services_response')
+        data_api_client.find_frameworks.return_value = {
+            'frameworks': [self.load_example_listing("framework_response")['frameworks']]
+        }
+
+        response = self.client.get('/admin/suppliers/1000/services?remove={}'.format(framework))
+        assert response.status_code == 200
+
+        document = html.fromstring(response.get_data(as_text=True))
+        expected_banner_message = "Are you sure you want to remove Supplier Name's 'G-Cloud 8' services?"
+        banner_message = document.xpath('//p[@class="banner-message"]//text()')[0].strip()
+        assert banner_message == expected_banner_message
+
+
+@mock.patch('app.main.views.suppliers.data_api_client')
+class TestDisableSupplierServicesView(LoggedInApplicationTest):
+
+    @pytest.mark.parametrize('url_suffix', ['', '?remove=', '?foo=bar'])
+    def test_400_if_no_framework_provided(self, data_api_client, url_suffix):
+
+        data_api_client.get_supplier.return_value = self.load_example_listing('supplier_response')
+        data_api_client.find_services.return_value = self.load_example_listing('services_response')
```

Comment:
In this scenario we never get as far as calling `find_services` - we can remove this mock value.

Actually in the POST route we never call `get_supplier`at all! Let's ditch that too 😸

Reply:
Nice catch! 👍 Done